mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -134,7 +138,11 @@ public void initialize(@NonNull Context context, @NonNull String accessToken, @N
     this.context = context.getApplicationContext();
     this.accessToken = accessToken;
     this.userAgent = userAgent;
-    if (this.context == null || TextUtils.isEmpty(this.accessToken) || TextUtils.isEmpty(this.userAgent)) {
+    this.sdkIdentifier = sdkIdentifier;
+    this.sdkVersion = sdkVersion;
+
+    if (this.context == null || TextUtils.isEmpty(this.accessToken) || TextUtils.isEmpty(this.userAgent)
+      || this.sdkIdentifier == null || this.sdkVersion == null) {
```

Comment:
👀 https://github.com/mapbox/mapbox-java/pull/574#issuecomment-331575490

> for now `null` values for the legacy Maps SDKs won't break anything.

`sdkIdentifier` and `sdkVersion` can't be empty either. We should mark somehow the different entry points (which `initialize` method was called) and make these checks wisely.